### ISSUE TITLE
Loss is 'Nan'

### ISSUE THREAD

**Issue by murdockhou** [slryou41/AnchorLoss#5]
Hi, thanks for your work.
 I used this `anchor loss` in my project to do classification job, I have tried set `warmup = False` and `warmup = True`, after several epochs, both loss become `Nan`. I have changed nothing in the source code `anchor_loss.py`. So, could you please give me some advice how can I fix it?  The loss when setting `warmup = True` like this:

![Selection_297](https://user-images.githubusercontent.com/18358653/84006584-90a85080-a9a1-11ea-9da1-f88b7abc717f.png)

Thanks a lot!

**Comment by slryou41** [comment]
Thanks for your interest! How many classes are you using? It looks like the loss is not decreasing even when you are using warmup, which is trained with the standard cross entropy. Were you able to train with cross entropy?

**Comment by murdockhou** [comment]
Hi, 3 classes in dataset. I can trained well with cross entropy. When I use cross entropy, the first several epochs maybe also learning nothing but with training process going, the network will learned something finally. So, it's worked with cross entropy but not worked with anchor loss by using same hyper parameters.

**Comment by slryou41** [comment]
This loss picks the prediction value from the target label and modulates the loss values for the other classes, so it might fluctuate when trained with 3 classes at the beginning. I still don't understand why it goes to NaN... Do you mind adding ``(1-pt).clamp(min=1e-10)`` in line 70 to make sure the loss never sees ``log(0)``?

**Comment by murdockhou** [comment]
No useful. I'm so sad. I trained model with `crossentropy` or `bce` all okay, only change the loss to `AnchorLoss`, after several epochs, got `nan`.......

**Comment by slryou41** [comment]
I guess the only reason why it goes to NaN is from ``log(0)``. Did you apply softmax or sigmoid before passing the output value to the loss function?  This loss function assumes that the output is not normalized.

**Comment by murdockhou** [comment]
@slryou41 No, only used original output from model, not normalized. The weired is the `corss entropy` or `bce` loss is okay but `anchor loss` not. So I don't think that there exists error about model's output.

**Comment by slryou41** [comment]
As a sanity check, can you set gamma = 0 and see if it goes to NaN?

**Comment by murdockhou** [comment]
Useless. Meet NaN same. I gave up this try...

**Comment by jongli747** [comment]
@slryou41  Hi. thanks for sharing your awesome work. I am trying to incorporate this loss in my project. I am facing the same "nan" problem after few epoch. I have followed everything you told earlier in this issue. But nothing works for me. so I am working on binary classification and output prediction is observed from the max operation. Please tell me what to do to get the work done.
